test/integration/appDuplicate.spec.ts

```diff
@@ -0,0 +1,34 @@
+import { test, expect, Page } from '../playwright/test';
+import generateId from '../utils/generateId';
+import * as locators from '../utils/locators';
+
+async function createApp(page: Page, name: string) {
+  await page.locator('button:has-text("create new")').click();
+
+  await page.fill('[role="dialog"] label:has-text("name")', name);
+
+  await page.click('[role="dialog"] button:has-text("create")');
+
+  await page.waitForNavigation({ url: /\/_toolpad\/app\/[^/]+\/editor\/pages\/[^/]+/ });
+}
+
+test('duplicate app from home flow', async ({ page }) => {
```

Comment:
we've created these page object models (https://playwright.dev/docs/test-pom) to try to encapsulate some common behaviors in different pages.
you can probably use the `ToolpadHome` model for some things in these tests?

Reply:
This looks very useful! For the app itself, do you see us making an AppCard/AppRow model as well to be able to capture the app-specific menu and buttons?

Reply:
> do you see us making an AppCard/AppRow model as well to be able to capture the app-specific menu and buttons?

Probably not. Why wouldn't `openApp(name)`, `editApp(name)`, `renameApp(name, newName)`,... work?

Reply:
Got it! I struggled a bit with making the test to check actual duplication work but can't seem to make it pass. I'll move it to a [separate PR](https://github.com/mui/mui-toolpad/tree/duplicate-app-test) and keep the minimal test which only checks whether the duplicated apps have been created.